### DISCUSSION TITLE
Can FSCrawler expand the ip of a network mapped drive?

### DISCUSSION THREAD

**Discussion by ghost** [dadoonet/fscrawler#1110] [Ideas]
I cant find reference to this in the docs, if there is i apologise please feel free to link me.

I have FSCrawler running and crawling a mapped drive X: when it ingests to fscrawler the file.url shows as:
file://X:\Folder...

Is it possible to have FSCrawler expand the drive IP and ingest that such that it would look like this:
file://10.1.1.1\Folder\Folder

The drive mapped X: is not very helpful to identify what drive is being crawler once its in elastic but with an ip i can find which machine the drive originates at.

Please let me know thought, if its not a feature could it be one?

Cheers

**Comment by dadoonet** [comment]
Could it be possible for you to use the `path.virtual` field instead and prefix it with `file://10.1.1.1` in your application?

There's a similar support though in the workplace search integration with https://fscrawler.readthedocs.io/en/latest/admin/fs/wpsearch.html#documents-repository-url. May be this setting should be moved in `fs` settings so we can use it also in elasticsearch output? WDYT?

Note that you would need to manually set it up. I don't see how FSCrawler could actually replace that automatically.

Last note. Why using `file://X:\Folder` as the `fs.url` instead of `file://10.1.1.1\Folder`?

**Reply by ghost** [reply]
Hi, Thanks for the reply,

Im not currently using any other applications I just run fscrawler to crawl my entire home media centre pc with its attached network drives I have 2. Theres no application between fscrawler and elastic nor after its been ingested into elastic i just search through that so no opportunity for me to append the ip address.

I wondered if there was perhaps a windows plugin that could be used by fscrawler when it crawls to resolve the drive name 'X' to the ip of the machine.

The workplace search integration i assume would only work for one of the mapped drive such that i set the 'url_prefix to '192.168.1.32' for say network drive 1, if i had a file (file.mp4) on network drive 2 '192.168.1.33' the url http://192.168.1.32/folder/file.mp4 would not be the location of the file as thats for drive 1. Hope that makes sense?

Im not sure what your last question is asking, im looking through the kibana interface and when a result is viewed in the properties it has file.url as 'file://X:\Folder' thats just what was entered by fscrawler, not modified by myself.

Cheers for the support.
